### PR TITLE
Sprint 6/issue 32# save recipes to favorites

### LINKED ISSUE #32: Save recipes to favorires
## REQUIREMENTS / DESCRIPTION

> Create functionality to save the recipes in favorites

- The recipe can be saved in the recipe item and details screen

```[tasklist]
### Tasks
- [ ] https://github.com/BrightCoders-Institute/proyecto-final-bcmay23-rn-t3-teamb/issues/31
```

### PR DESCRIPTION
**Pull request by AMCernas**
# Description

What did you implemented/Why did you implemented this:

- Users now can save recipes in favorites.
- FavoritesScreen now shows the current user favorite recipes.
- Icon to save recipes in favorites works dynamically.
- The user can also see the recipe details, in FavoritesScreen.
- popularRecipe component in Homescreen now works and can show a recipe, for now, this recipe is random.

## Screenshots

image
image

image

### CODE REVIEW

File: src/components/PopularRecipe/PopularRecipe.jsx [hunk 1]
```diff
@@ -7,24 +7,23 @@ import Modal from 'react-native-modal';
...
+  const formattedServings = servings !== undefined ? servings.toString() : '';
+  const recipeTitle = recipeData[titleKey];
+  const summaryWithoutTags = recipeData.summary ? recipeData.summary.replace(/<b>|<\/b>/gi, '') : '';
+  const match = summaryWithoutTags.match(/^(.*?)(?=It is brought to you by Foodista)/s);
+  const caloriesDescription = match && match[1] ? match[1] : summaryWithoutTags;
+
   const toggleModal = () => {
     setModalVisible(!isModalVisible);
   };
+  console.log(recipeData)
```
Comment (ecruz17) [inline src/components/PopularRecipe/PopularRecipe.jsx:22]: just be careful with leaving logs ;)

File: src/database/index.ts [hunk 2]
```diff
@@ -5,9 +5,11 @@ import envs from '../config/env';
 const {API_KEY, API_HOST} = envs;
 
 export const API = axios.create({
-  baseURL: 'https://spoonacular-recipe-food-nutrition-v1.p.rapidapi.com/recipes',
+  baseURL: 'https://api.spoonacular.com/recipes',
+  params: {
+    apiKey: "0a190ca3a52b45a59e83b457a538e2b9",
+  },
```
Comment (ecruz17) [inline src/database/index.ts:11]: I'd suggest you for safety purpouses to hide this sensitive data from the public repo

**Review by ecruz17 — APPROVED** [review]
Nice work! Just consider those comments for a more robust product ;)

**Review by Lizzflores — APPROVED** [review]

File: src/database/index.ts (hunk 2 above)
Comment (AMCernas) [reply src/database/index.ts:11]: oh, i forgot about that, thnxs

**Review by ShadowEnchanter — APPROVED** [review]

**Pull request merged**